### PR TITLE
fixed out of range access in FindRampNDIndex

### PR DESCRIPTION
**Pull request by MiraiHattori**
the code path may access `*_rampnds.end()`, so fixing.

### CODE REVIEW

**Comment by MiraiHattori** [comment]
note: pipeline id is 558171

File: plugins/rplanners/rampoptimizer/ramp.cpp [hunk 1]
```diff
@@ -989,7 +989,7 @@ void ParabolicPath::FindRampNDIndex(dReal t, int& index, dReal& remainder) const
         int index_ = 0;
         dReal curTime = 0;
         std::vector<RampND>::const_iterator itrampnd = _rampnds.begin();
-        while ((itrampnd - 1) != _rampnds.end() && t >= curTime) {
+        while( itrampnd != _rampnds.end() && t >= curTime ) {
```
Comment (kanbouchou) [inline plugins/rplanners/rampoptimizer/ramp.cpp:992]: There is similar code in ``ParabolicCurve::FindRampIndex``, please check that as well.

Comment (MiraiHattori) [reply plugins/rplanners/rampoptimizer/ramp.cpp:992]: @kanbouchou that's true. fixed `PiecewisePolynomial::FindPolynomialIndex` as well. could you review them?

Comment (rdiankov) [reply plugins/rplanners/rampoptimizer/ramp.cpp:992]: @Puttichai  can you check?

Comment (Puttichai) [reply plugins/rplanners/rampoptimizer/ramp.cpp:992]: Looks good to me.

Comment (yoshikikanemoto) [reply plugins/rplanners/rampoptimizer/ramp.cpp:992]: looks good to me too

**Comment by rdiankov** [comment]
thanks~

**Pull request merged**